### PR TITLE
feat: add a way to request user script callback.

### PR DESCRIPTION
**Pull request by schinkowski**

### CODE REVIEW

File: Runtime/Scripts/MeshSyncServer.Properties.cs [hunk 1]
```diff
@@ -141,9 +141,14 @@ internal enum PropertiesState {
 #endif
             }
 
-            if (needsClientSync) {
+            if (needsUserScriptCallback) {
+                MeshSyncLogger.VerboseLog("Sending changes, needed user script callback.");
+                needsUserScriptCallback = false;
+                sendChanges             = true;
+                m_server.RequestUserScriptCallback();
```
Comment (SonnyCampbellUnity) [inline Runtime/Scripts/MeshSyncServer.Properties.cs:148]: Does this set `needsUserScriptCallback` to false, and then immediately call `RequestUserScriptCallback` which sets it back to true again?

File: Runtime/Scripts/MeshSyncServer.Properties.cs [hunk 2]
```diff
@@ -262,5 +267,11 @@ protected override void SplineChanged(Spline spline, int arg2, SplineModificatio
         needsClientSync = true;
     }
 #endif
+
+
+        internal void RequestUserScriptCallback() {
+            needsUserScriptCallback = true;
+        }
+#endif
```
Comment (seandillon92) [inline Runtime/Scripts/MeshSyncServer.Properties.cs]: where is the opening directive? I think this is causing compilation errors in yamato, all test are failing

File: Runtime/Scripts/MeshSyncServer.Properties.cs (hunk 1 above)
Comment (schinkowski) [reply Runtime/Scripts/MeshSyncServer.Properties.cs:148]: No, this tells the C++ server to send a message to the client to call a user script.

**Review by seandillon92 — APPROVED** [review]

File: Runtime/Scripts/MeshSyncServer.Properties.cs (hunk 1 above)
Comment (sindharta) [reply Runtime/Scripts/MeshSyncServer.Properties.cs:148]: Could you please remind me if there is a diagram somewhere that shows the flow of this user script callback ?

Comment (schinkowski) [reply Runtime/Scripts/MeshSyncServer.Properties.cs:148]: There's no diagram, this just sends the "user_script_callback" message from the server to the client, which triggers the blender addon to call any methods added to "mesh_sync_server_requested_callback" in the addon. Users can add their own calls to that using an addon or python code in a blend file.

**Review by sindharta — APPROVED** [review]

**Review by SonnyCampbellUnity — APPROVED** [review]

**Pull request merged**